src/main/java/be/seeseemelk/mockbukkit/MockUnsafeValues.java

```diff
@@ -296,32 +299,68 @@ public Entity deserializeEntity(byte[] data, World world, boolean preserveUUID)
 		throw new UnimplementedOperationException();
 	}
 
+	/**
+	 * Gets the translation key for a {@link Material} that {@link Material#isBlock() is a block}.
+	 * @param material the material to translate
+	 * @return a string of the structure "block.{@literal <namespace>}.{@literal <material>}" (e.g. "block.minecraft.stone"), or null if not a block.
+	 */
 	@Override
 	public String getBlockTranslationKey(Material material)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if(!material.isBlock()) {
+			return null;
+		}
+		return String.format("block.%s.%s", material.key().namespace(), material.key().value());
 	}
 
+	/**
+	 * Gets the translation key for a {@link Material} that {@link Material#isItem() is an item}.
+	 * @param material the material to translate
+	 * @return a string of the structure "item.{@literal <namespace>}.{@literal <material>}" (e.g. "item.minecraft.carrot"), or null if not an item.
+	 */
 	@Override
 	public String getItemTranslationKey(Material material)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if(!material.isItem()) {
+			return null;
+		}
+		return formatTranslatable("item", material);
 	}
 
+	/**
+	 * Gets the translation key for an {@link EntityType}. Throws an error for custom entities.
+	 * @param type the entity to translate
+	 * @return a string of the structure "item.{@literal <namespace>}.{@literal <entity_type>}" (e.g. "entity.minecraft.pig").
+	 */
 	@Override
 	public String getTranslationKey(EntityType type)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkArgument(type.getName() != null, "Invalid name of EntityType %s for translation key", type);
+		Arrays.stream(EntityType.values())
+				.filter(entityType ->type.getName().equals(entityType.getName()))
+				.findFirst()
+				.orElseThrow();
+		return formatTranslatable("entity", type);
 	}
 
+	/**
+	 * Gets the translation key for a {@link ItemStack} that {@link Material#isItem() is an item}.
+	 * @param itemStack the itemstack to translate
+	 * @return a string of the structure "item.{@literal <namespace>}.{@literal <material>}" (e.g. "item.minecraft.carrot"), or null if not an item.
+	 */
 	@Override
 	public String getTranslationKey(ItemStack itemStack)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if(!itemStack.getType().isItem()) {
+			return null;
+		}
+		return formatTranslatable("item", itemStack.getType());
```

Comment:
I'm unsure how to handle translation keys for ItemStacks. I don't know how paper handles translation of custom items...

Reply:
You don't necessarily need to implement this, you can alternatively throw UnimplementedOperationException